Modules/Core/Common/include/itkByteSwapper.hxx

```diff
@@ -44,49 +45,19 @@ template <typename T>
 void
 ByteSwapper<T>::SwapFromSystemToBigEndian([[maybe_unused]] T * p)
 {
-  if constexpr (!m_SystemIsBigEndian)
+  if constexpr (!m_SystemIsBigEndian && sizeof(T) > 1)
   {
-    switch (sizeof(T))
-    {
-      case 1:
-        return;
-      case 2:
-        Self::Swap2(p);
-        return;
-      case 4:
-        Self::Swap4(p);
-        return;
-      case 8:
-        Self::Swap8(p);
-        return;
-      default:
-        itkGenericExceptionMacro("Cannot swap number of bytes requested");
-    }
+    SwapBytes(*p);
   }
 }
 
 template <typename T>
 void
 ByteSwapper<T>::SwapRangeFromSystemToBigEndian([[maybe_unused]] T * p, [[maybe_unused]] BufferSizeType num)
 {
-  if constexpr (!m_SystemIsBigEndian)
+  if constexpr (!m_SystemIsBigEndian && sizeof(T) > 1)
   {
-    switch (sizeof(T))
-    {
-      case 1:
-        return;
-      case 2:
-        Self::Swap2Range(p, num);
-        return;
-      case 4:
-        Self::Swap4Range(p, num);
-        return;
-      case 8:
-        Self::Swap8Range(p, num);
-        return;
-      default:
-        itkGenericExceptionMacro("Cannot swap number of bytes requested");
-    }
+    std::for_each_n(p, num, [](T & element) { SwapBytes(element); });
```

Comment:
Forgive me if this is not possible, but this block is the same as in https://github.com/InsightSoftwareConsortium/ITK/pull/4855/files#diff-097dc3a747bd453b8fcfb3f846d65f65931572c23656e478ea40e591a4576bbcR113, so wondering if a private method can be created to be call from both functions.

Reply:
Sure, I can make a private `SwapRange` helper function to avoid having duplicate `std::for_each_n` call. 👍

Reply:
Additional level of indirection to save duplication of one line/function call might be counterproductive to understanding the code.

Reply:
> Additional level of indirection to save duplication of one line/function call might be counterproductive to understanding the code.

Thanks to clang-format that it is just one line! It might as well have been three lines of code:

```cpp
std::for_each_n(p, num, [](T & element)
  {
    SwapBytes(element);
  });
```

@jhlegarreta I see your point, and yes, technically a helper function is possible, but is it still acceptable for you to leave it as-is?

Reply:
Acceptable.